code/WorkInProgress/zamu_mail.dm

```diff
@@ -269,23 +327,91 @@
 
 		// make a gift for this person
 		var/obj/item/random_mail/package = null
-		var/package_color = "#FFFFFF"
+		var/package_color = recipient["fav_color"] ? recipient["fav_color"] : pick("#FFFFAA", "#FFBB88", "#FF8800", "#CCCCFF", "#FEFEFE")
+		var/stamp_id = "stamp-stain-[rand(1, 3)]"
+		var/sender = null
+		var/sender_title = null
+		var/list/context = list()
+
+		// chance to find an appropriate mail based on the traits chosen
+		if (prob(percentmult(20, length(picked.traitHolder.traits)) * 0.6))
+			var/list/possible_mail = list()
+			for (var/datum/trait in picked.traitHolder.traits)
+				var/typeinfo/datum/trait/typeinfo = trait.get_typeinfo()
+				if (length(typeinfo.mail_items > 0))
+					var/random_pick = pick(typeinfo.mail_items[trait])
```

Comment:
This will result in the last valid trait always being picked to spawn mail, if you want it to be even you will need to build a list of valid traits and pick from it, then pick a piece of mail to spawn from that trait's typeinfo.